### PR TITLE
feat(client): create supplier interface

### PR DESCRIPTION
**Pull request by AzizSandid**
## What does this do?

create supplier interface
make some changes 
add translation

### CODE REVIEW

File: src/services/shared/supplier-services.ts [hunk 1]
```diff
@@ -110,3 +113,23 @@ export async function updateSocialLinkAsync(
...
   return client.put(url, formData);
 }
+
+export async function getAllSuppliers(): Promise<SupplierProfileInfoListType> {
+  const url = '/api/users/supplier/get-all-suppliers/';
+  return client
+    .get(url)
+    .then((response) => response.data)
+    .catch((error) => {
+      if (isAxiosError(error)) {
```
Comment (MohamedMegdich1) [inline src/services/shared/supplier-services.ts:123]: Please make a refacto this API the error handling

File: src/modules/client/supplier/hooks/use-supplier.tsx [hunk 2]
```diff
@@ -24,11 +33,43 @@ export const useSupplier = () => {
...
+    isSuccess: isSuccessSuppliers,
+  } = useAllSuppliersApi();
+
+  const modal = useModal();
+  const cityOptions = VALID_CITIES.map((city) => ({
+    label: city,
+    value: city,
+  }));
+  const { control: controlCity } = useCustomForm(cityTypeSchema, {
+    city: 'Tunis',
```
Comment (MohamedMegdich1) [inline src/modules/client/supplier/hooks/use-supplier.tsx:53]: Why are you using static values?

File: src/modules/client/supplier/hooks/use-supplier.tsx [hunk 3]
```diff
@@ -24,11 +33,43 @@ export const useSupplier = () => {
         imageIcon: type.icon,
       };
     }) || [];
-
   const { control } = useCustomForm(specialityTypeSchema, {
     specialityType: 'Matériaux de construction',
```
Comment (MohamedMegdich1) [inline src/modules/client/supplier/hooks/use-supplier.tsx:37]: the same

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: src/modules/client/supplier/hooks/use-supplier.tsx (hunk 2 above)
Comment (AzizSandid) [reply src/modules/client/supplier/hooks/use-supplier.tsx:53]: image
To ensure the correct content is displayed on the initial render, I set the first selection as the default value. This way, the supplier cards below reflect the correct data based on the default selected value, providing a seamless user experience.

File: src/modules/client/supplier/hooks/use-supplier.tsx (hunk 3 above)
Comment (AzizSandid) [reply src/modules/client/supplier/hooks/use-supplier.tsx:37]: same

File: src/services/shared/supplier-services.ts (hunk 1 above)
Comment (AzizSandid) [reply src/services/shared/supplier-services.ts:123]: fixed

File: src/modules/client/supplier/hooks/use-supplier.tsx [hunk 4]
```diff
@@ -1,12 +1,21 @@
...
+import type { Option } from '@/shared/components';
+import { useModal } from '@/shared/components';
 import type { TagType } from '@/types';
 
 import {
   CategoryData,
   PropertyData,
 } from '../../create-announcement/dump-data';
 import { specialityTypeSchema } from '../schema/speciality-schema';
+import { cityTypeSchema } from '../schema/speciality-schema copy';
```
Comment (MohamedMegdich1) [inline src/modules/client/supplier/hooks/use-supplier.tsx:18]: the naming of the file

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: src/modules/client/supplier/hooks/use-supplier.tsx (hunk 4 above)
Comment (AzizSandid) [reply src/modules/client/supplier/hooks/use-supplier.tsx]: fixed

**Review by MohamedMegdich1 — APPROVED** [review]

**Pull request merged**